source/base/parameter_handler.cc

```diff
@@ -519,17 +519,25 @@ ParameterHandler::parse_input(const std::string &filename,
       std::istream is(&fb);
       std::string file_ending = filename.substr(filename.find_last_of('.') + 1);
       boost::algorithm::to_lower(file_ending);
-      if (file_ending == "prm")
-        parse_input(is, filename, last_line, skip_undefined);
-      else if (file_ending == "xml")
+
+      if (file_ending == "xml")
         parse_input_from_xml(is, skip_undefined);
       else if (file_ending == "json")
         parse_input_from_json(is, skip_undefined);
       else
-        AssertThrow(
-          false,
-          ExcMessage(
-            "Unknown input file. Supported types are .prm, .xml, and .json."));
+        {
+          // See: discussion in https://github.com/dealii/dealii/pull/9741
+          // This PR contains the code which we should use from release
+          // 9.3 onwards.
```

Comment:
I would like to add a comment explaining what we are doing here: "File endings other than prm, xml, json are currently interpreted as files of prm-type for reasons of backward compatibility."